### PR TITLE
refactoring activity

### PR DESCRIPTION
**Pull request by mdmahmudurrahman**

### CODE REVIEW

File: app/controllers/favourites_controller.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@ class FavouritesController < ApplicationController
 
   def update
     @user_book = UserBook.find params[:id]
-    FavouriteService.new(@user_book).update book_params
+    FavouriteService.new(@user_book, current_user).update book_params
+    @user_book.create_activity key: "UsersBook.favourite", owner: current_user
```
Comment (MuhammadTamzid) [inline app/controllers/favourites_controller.rb]: push this code on service

File: app/services/favourite_service.rb [hunk 2]
```diff
@@ -1,13 +1,22 @@
 class FavouriteService
-  def initialize user_book
+  def initialize user_book, user
```
Comment (MuhammadTamzid) [inline app/services/favourite_service.rb]: def initialize user_book, user=nil

File: app/views/books/index.html.erb [hunk 3]
```diff
@@ -14,7 +14,7 @@
         <td>
           <div class="row">
             <div class="col-md-2">
-              <%= link_to image_tag(book.book_photos.first.image.url(:book_medium)), book %>
+            <%= link_to image_tag(book.book_photos.first.image.url(:book_medium)), book %>
```
Comment (shakil-fg) [inline app/views/books/index.html.erb]: restore indentation

Comment (MuhammadTamzid) [inline app/views/books/index.html.erb]: indent

**Comment by shakil-fg** [comment]
ok

**Pull request merged**